addons/web_editor/static/src/js/wysiwyg/widgets/link_dialog.js

```diff
@@ -51,19 +56,17 @@ var LinkDialog = Dialog.extend({
         ];
 
         this.editable = editable;
-        this.data = linkInfo || {};
-
         this.data.className = "";
         this.data.iniClassName = "";
 
         var r = this.data.range;
         this.needLabel = !r || (r.sc === r.ec && r.so === r.eo);
 
         if (this.data.range) {
-            const $link = $(this.data.range.sc).filter("a");
-            this.data.iniClassName = $link.attr("class") || "";
+            const $el = $(this.data.range.sc).filter(this.isButton ? "button" : "a");
+            this.data.iniClassName = $el.attr("class") || "";
             this.colorCombinationClass = false;
-            let $node = $link;
+            let $node = $el;
```

Comment:
This old code here is checking the DOM according to the given range... so I don't think it should depend on the `isButton` config which might not make sense: I think you have to filter in search for button **or** link and adapt the `isButton` value accordingly.

Reply:
Actually, The goal is not to update the value of "isButton"... we check if the target is a button element in Editor's "getLinkInfo" method so we can use it here to get the "iniClassName" of the right element (anchor or button).
This way we have the style in preview for buttons too...

Reply:
Why do we need to search for the button or the link if we already found it before then? But ok, I guess this is old code we should not refactore in stable version anyway...

Reply:
The minor difference here is about `iniClassName`:
if we keep the `filter("a")`, we'll get the initial style only for anchor and button will appear in preview with no class applied.